ui/src/components/StoryDrawer.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from "react";
+import Drawer from "@material-ui/core/Drawer";
+import {
+  DrawerAuthorText,
+  DrawerContentText,
+  DrawerCountryText,
+  DrawerRightText,
+  DrawerTitleText,
+} from "../styles/typography";
+import ArrowForwardIcon from "@material-ui/icons/ArrowForward";
+import Button from "@material-ui/core/Button";
+import Grid from "@material-ui/core/Grid";
+import styled from "styled-components";
+import ChinaFlag from "../assets/flags/China.png";
+import TempShoe from "../assets/images/temp.png";
+import { colors } from "../styles/colors";
+
+const StyledIconButton = styled(Button)`
+  position: absolute;
+  left: 0px;
+  top: 0px;
+  && {
+    background-color: ${colors.secondary};
+    color: ${colors.white};
+    min-width: 66.77px;
+    min-height: 60px;
+    width: 66.77px;
+    height: 60px;
+    border-radius: 0px;
+    &:hover {
+      background-color: ${colors.secondaryDark1};
+    }
+  }
+`;
+
+const StyledImage = styled.img`
+  border-radius: 0px;
+  width: 100%;
+`;
+
+const StyledRoot = styled.div`
+  width: 100vh;
+  padding-left: 30vh;
+  padding-right: 30vh;
+`;
+
+const StyledFlagImage = styled.img`
+  border-radius: 10px;
+  width: 100%;
+`;
+
+export enum StoryDrawerState {
+  Open,
+  Closed,
+}
+
+interface StoryDrawerProps {
+  title: string;
+  author: string;
+  date: string;
+  country: string;
+  content: string;
+  currentCity: string;
+  state: StoryDrawerState;
+  setState: (newState: StoryDrawerState) => void;
+}
+
+export function StoryDrawer({
+  title,
+  author,
+  date,
+  country,
+  currentCity,
+  content,
+  state,
+  setState,
+}: StoryDrawerProps): JSX.Element {
+  return (
+    <Drawer
+      anchor="right"
+      open={state === StoryDrawerState.Open}
+      onClose={() => setState(StoryDrawerState.Closed)}
+    >
+      <StyledIconButton onClick={() => setState(StoryDrawerState.Closed)}>
+        <ArrowForwardIcon />
```

Comment:
This is super tiny but I think the arrow in Figma isn't MaterialUI (it's a bit longer) but I don't think that matters.

Reply:
Just left this the same